File: C++ for Everyone, 2nd Ed, Horstmann/Chapter4/P4-3D.cpp
//Write programs that read a line of input as a string and print
//d.The number of vowels in the string.

#include<iostream>
#include<vector>

using namespace std;

int main()
{
    cout << "Enter a string: ";
    string input;
    getline(cin, input);

    vector<char> inputCstyle (input.length() + 1);
    copy(input.begin(), input.end(), inputCstyle.begin());

    int nVowels = 0;
    for(int i = 0; i < input.length(); i++)
    {
        if(inputCstyle[i] == 'A' || inputCstyle[i] == 'a' || inputCstyle[i] == 'E' || inputCstyle[i] == 'e' || inputCstyle[i] == 'O' || inputCstyle[i] == 'o' || inputCstyle[i] == 'U' || inputCstyle[i] == 'u' || inputCstyle[i] == 'I' || inputCstyle[i] == 'i')
        {
            nVowels++;
        }
    }

    cout << "The numbe of vowels in <" << input << "> is: " << nVowels;
    return 0;
}

File: Python For Everyone, Horstmann/Chapter4/P4.3D.py
# Write programs that read a line of input as a string and print
#     a.  Only the uppercase letters in the string.
#     b.  Every second letter of the string.
#     c.  The string, with all vowels replaced by an underscore.
#     d.  The number of digits in the string.
#     e.  The positions of all vowels in the string.


inputString = str(input("Enter a string: "))
numDigits = 0

for i in range(len(inputString)):
    if inputString[i].isdigit():
        numDigits += 1

print("Number of digits:", numDigits)